### PR TITLE
Security docs

### PR DESCRIPTION
**Pull request by mtremsal**
### What does this PR do?
This PR introduces a set of security documentation pages that consolidate features and guarantees for each product line.

### Motivation
Customer questions are regularly submitted to support@ and security@ mailboxes in regards to security features. Customers expect security-related documentation to be clearly accessible in a single place.

### Preview link
[staging](https://docs-staging.datadoghq.com/marc.tremsal/security-docs/security/)

### Additional Notes
Feedback from @cswatt and @l0k0ms has been addressed, with the exception of 'Metadata' which is still capitalized due to being defined as its own category of data. 

Updates to these pages *require* a formal review by Datadog Security.

### CODE REVIEW

File: content/security/tracing.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+Several filtering mechanisms are enforced as a baseline in an effort to provide sound defaults. In particular:
+
+**Environment variables are not collected by the Agent**
+
+**SQL variables are obfuscated by default, even when not using prepared statements**
+
+For example, the following `sql.query` attribute:
+`SELECT data FROM table WHERE key=123 LIMIT 10`
+would have its variables obfuscated, to become the following Resource name:
+`SELECT data FROM table WHERE ? LIMIT ?`
```
Comment (palazzem) [inline content/security/tracing.md]: `SELECT data FROM table WHERE key = ? LIMIT ?`

File: content/security/agent.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+If you believe you’ve discovered a bug in Datadog’s security, please get in touch at [security@datadoghq.com][8] and we will get back to you within 24 hours. Our [PGP key][9] is available for download in case you need to encrypt communications with us. We request that you not publicly disclose the issue until we have had a chance to address it.
+
+## [BETA] Secrets Management
+
+Customers with a requirement to avoid storing secrets in plaintext in the Agent's configuration files can leverage the [secrets management][10] package. This package allows the Agent to call a user-provided executable to handle retrieval or decryption of secrets, which are then loaded in memory by the Agent. Users have the flexibility to design their executable according to their preferred key management service, authentication method, and continuous integration workflow.
```
[the remark this answers is not in the document]
Comment (mtremsal) [reply content/security/agent.md]: Yes, eventually I want to expand this section. Maybe we can keep explanations light for the duration of the beta, and revisit this language once GA.

File: content/security/agent.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+This article is part of a [series on data security][0].
+
+Customers can send data to the Datadog service by using a locally installed [Agent][1] or through our [HTTP API][2]. While use of Datadog does not strictly require use of the Datadog Agent, the vast majority of customers leverage the Agent. This article describes the main security capabilities and features available to customers to ensure their environment is secure.
+
+## Data Protection
+
+The Datadog Agent submits data to Datadog over a TLS-encrypted TCP connection by default. As of version 6, the Agent can be configured to enforce TLS 1.2 when connecting to Datadog. Customers who require the use of "strong cryptography," for example, to meet PCI requirements, should use Agent v6 and set the `force_tls_12: yes` setting in the Agent’s configuration file.
+
+## Agent Logs Obfuscation
```
[the remark this answers is not in the document]
Comment (mtremsal) [reply content/security/agent.md]: I've tweaked the language to reflect this: 'local logs are filtered for *some specific* keywords'

**Comment by jeremy-lq** [comment]
@mtremsal, looks like we have approval from each team. I've approved, feel free to merge. If you need anything else, let us know.

File: content/security/agent.md [hunk 4]
```diff
@@ -19,15 +19,15 @@ The Datadog Agent submits data to Datadog over a TLS-encrypted TCP connection by
 
 ## Agent Logs Obfuscation
 
-The Datadog Agent generates local logs in order to support [Agent troubleshooting][3] as required. As a safety precaution, these local logs are filtered for keywords that could indicate a potential credential (e.g. API key, password and token keywords, etc.), which are then obfuscated before being written to disk.
+The Datadog Agent generates local logs in order to support [Agent troubleshooting][3] as required. As a safety precaution, these local logs are filtered for some specific keywords that could indicate a potential credential (e.g. API key, password and token keywords, etc.), which are then obfuscated before being written to disk.
```
Comment (truthbk) [inline content/security/agent.md]: ideally specific _keywords and patterns_

**Comment by mtremsal** [comment]
@jeremy-lq Legal has just approved this version! 🎉 
I don't have write access to this repo, could you please merge the PR? 🙇

**Pull request merged**